File: c/Java/Symbol.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.stream.Collectors;

public class Symbol {
    private String name;
    private HashSet<TypeClassification> classification;
    private Map<String, Symbol> members = new HashMap<>();
    private Symbol parent;
    private boolean predefined = false;
    private String definedFile = "";
    private int definedLine = 0;
    private int definedColumn = 0;

    public String getDefinedFile() {
        return definedFile;
    }

    public void setDefinedFile(String definedFile) {
        this.definedFile = definedFile;
    }

    public int getDefinedLine() {
        return definedLine;
    }

    public void setDefinedLine(int definedLine) {
        this.definedLine = definedLine;
    }

    public int getDefinedColumn() {
        return definedColumn;
    }

    public void setDefinedColumn(int definedColumn) {
        this.definedColumn = definedColumn;
    }

    public boolean isPredefined() {
        return predefined;
    }

    public void setPredefined(boolean predefined) {
        this.predefined = predefined;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public HashSet<TypeClassification> getClassification() {
        return classification;
    }

    public void setClassification(HashSet<TypeClassification> classification) {
        this.classification = classification;
    }

    public Map<String, Symbol> getMembers() {
        return members;
    }

    public Symbol getParent() {
        return parent;
    }

    public void setParent(Symbol parent) {
        this.parent = parent;
    }

    @Override
    public String toString() {
        String result = name;
        String classificationStr = classification.stream()
            .map(TypeClassification::name)
            .collect(Collectors.joining(", "));
        result += " (with classification " + classificationStr + ")";
        if (definedFile != null && !definedFile.isEmpty()) {
            result += " defined at " + definedFile + ":" + definedLine + ":" + definedColumn;
        }
        return result;
    }
}

File: c/Go/Symbol.go
package parser

import (
	"fmt"
	"strings"
)

type Symbol struct {
	Name           string
	Classification map[TypeClassification]bool
	Members        map[string]*Symbol
	Parent         *Symbol
	Predefined     bool
	DefinedFile    string
	DefinedLine    int
	DefinedColumn  int
}

func newSymbol(name string, classifications ...TypeClassification) *Symbol {
	s := &Symbol{
		Name:           name,
		Classification: make(map[TypeClassification]bool),
		Members:        make(map[string]*Symbol),
	}
	for _, c := range classifications {
		s.Classification[c] = true
	}
	return s
}

func newPredefinedSymbol(name string, classifications ...TypeClassification) *Symbol {
	s := newSymbol(name, classifications...)
	s.Predefined = true
	return s
}

func (s *Symbol) String() string {
	names := make([]string, 0, len(s.Classification))
	for c := range s.Classification {
		names = append(names, fmt.Sprintf("%d", int(c)))
	}
	result := s.Name + " (with classification " + strings.Join(names, ", ") + ")"
	if s.DefinedFile != "" {
		result += fmt.Sprintf(" defined at %s:%d:%d", s.DefinedFile, s.DefinedLine, s.DefinedColumn)
	}
	return result
}
